curriculum/challenges/english/07-scientific-computing-with-python/learn-tree-traversal-by-building-a-binary-search-tree/65c646d4148ae3b2d1cbcac4.md

```diff
@@ -11,12 +11,14 @@ Note that, your search returns something like `80: <__main__.TreeNode object at
 
 To change that to print a useful value, define another method named `__str__` in the `TreeNode` class. It takes a single argument `self`.
 
+After defining `__str__` you'll get an exception in the console because the `__str__` method doesn't return anything yet. You'll work on the method body in the next step.
+
 # --hints--
 
 You should define a method `__str__` that takes a single argument `self`. Remember to use `pass`.
 
 ```js
-assert.match(code, /def\s+__str__\(\s*self\s*\)/);
+assert.match(code, /^(\s+)def\s+__init__.+?\1def\s+__str__\(\s*self\s*\)\s*:\s*\n\1\1pass/ms)
```

Comment:
```suggestion
assert.match(code, /^(\s+)def\s+__init__.+?^\1def\s+__str__\(\s*self\s*\)\s*:\s*\n^\1\1pass/ms)
```

maybe? the other version wants at least 4 spaces, but it doesn't check that it is start of line + four spaces.